File: src/go/test/testml-bridge.go
package bridge

import "os"

type Bridge struct { }

type Mine struct { }

func New() *Bridge {
  self := new(Bridge)
  return self
}

func (self *Bridge) Hash_lookup(
  hash map[string]interface{},
  key string,
) interface{} {

  return hash[key]
}

func (self *Bridge) Get_env(name string) string {
  return os.Getenv(name)
}

func (self *Bridge) Add(x, y int) int {
  return x + y
}

func (self *Bridge) Sub(x, y int) int {
  return x - y
}

func (self *Bridge) Cat(x, y string) string {
  return x + y
}

func (self *Bridge) Mine() *Mine {
  return new(Mine)
}

File: src/ruby/test/testml-bridge.rb
# require 'testml/bridge'
class TestMLBridge # < TestML::Bridge

# sub hash_lookup {
#   my ($self, $hash, $key) = @_;
#   $hash->{$key};
# }
# 
# sub get_env {
#   my ($self, $name) = @_;
#   $ENV{$name};
# }

  def add(x, y)
    return x + y
  end

  def sub(x, y)
    return x - y
  end

# sub cat {
#   my ($self, $x, $y) = @_;
# 
#   $x . $y;
# }
# 
# sub mine {
#   bless {}, 'Mine';
# }

end

# vim: set sw=2 sts=2 et:
